experimental/dds/tree2/src/shared-tree/sharedTreeView.ts

```diff
@@ -468,12 +483,20 @@ export class SharedTreeView implements ISharedTreeView {
 		view.rebase(this);
 	}
 
-	public merge(fork: SharedTreeView): void {
-		this.branch.merge(fork.branch);
-	}
-
-	public rebase(fork: SharedTreeView): void {
-		fork.branch.rebaseOnto(this.branch);
+	public merge(view: SharedTreeView): void;
+	public merge(view: SharedTreeView, disposeView: boolean): void;
+	public merge(view: SharedTreeView, disposeView = true): void {
```

Comment:
It's not necessary, but it makes the `disposeView` parameter slightly less discoverable. When somebody gets intellisense for the method they'll only see the simple one without `disposeView`. I like this because we want users to avoid using the "long running" (i.e. not auto-disposed) branches unless they really know what they are doing (lest they forget to dispose it).